R/R/prophet.R

```diff
@@ -1274,9 +1288,134 @@ fit.prophet <- function(m, df, ...) {
+    t = object$history$t,                                
+    cap = rep(0, nrow(object$history)),                         
+    t_change = array(object$changepoints.t),
+    X = as.matrix(seasonal.features[1:nrow(object$history),]),            
+    sigmas = array(out2$prior.scales),
+    tau = object$changepoint.prior.scale,
+    trend_indicator = as.numeric(object$growth == 'logistic'),
+    s_a = array(component.cols$additive_terms),
+    s_m = array(component.cols$multiplicative_terms),
+    
+    T_pred = nrow(history.and.future),
+    t_pred = history.and.future$t,
+    cap_pred = 
+      if(class(history.and.future$cap) == "NULL") { 
+        rep(0, length(history.and.future$t)) 
+      } else { 
+        history.and.future$cap 
+      },
+    X_pred = as.matrix(seasonal.features),
+    n_samp = object$uncertainty.samples
+  )
+  
+  # Future # of change points assuming a homog. Poisson process
+  dat$S_pred <- stats::qpois(0.95, dat$S * (dat$T_pred - dat$T) / dat$T )
+  
+  # Run stan
+  if (exists(".prophet.stan.model")) {
+    model <- .prophet.stan.model
+  } else {
+    model <- get_prophet_stan_model()
+  }
+  
+  stan_init <- 
+    list(
+      k = object$params$k,
+      m = object$params$m,
+      delta = as.numeric(object$params$delta),
+      beta = as.numeric(object$params$beta),
+      sigma_obs = object$params$sigma_obs
+    )
+  
+  if (m$mcmc.samples > 0) {
+    args <- list(
+      object = model,
+      data = dat,
+      init = stan_init,
+      iter = m$mcmc.samples
+    )
+    args <- utils::modifyList(args, list(...))
+    stan.fit <- do.call(rstan::sampling, args)
+  } else {
+    args <- list(
+      object = model,
+      data = dat,
+      init = stan_init,
+      iter = 0,
+      as_vector = FALSE,
+      verbose = TRUE
+    )
+    args <- utils::modifyList(args, list(...))
+    stan.fit <- do.call(rstan::optimizing, args)
+    
+    # Extract overall and trend predictions
+    stan.predictions <- 
+      stan.fit$par[c("y_hat", "trend_hat")] %>%
+      as.data.frame() %>%
+      mutate(yhat = y_hat, trend = trend_hat)
+    
+    # Extract overall and trend intervals
+    stan.intervals <- predict_uncertainty(object, stan.fit)
+  }
+  
+  
+  
+  df <- 
+    cbind(
+      ds = history.and.future$ds, 
+      stan.predictions, 
+      stan.intervals
+    ) %>% 
+    select( # Retain only overall and trend predictions
+      ds, 
+      yhat, yhat_lower, yhat_upper, 
+      trend, trend_lower, trend_upper
+    ) %>% 
+    mutate_at( # Rescale overall and trend predictions and intervals
+      c(
+        "yhat", "yhat_lower", "yhat_upper",
+        "trend", "trend_lower", "trend_upper"
+      ),
+      funs(. * object$y.scale + history.and.future$floor)
+    ) %>% # Now add seasonal breakdowns and intervals
+    cbind(
+      predict_seasonal_components(object, history.and.future)
+    )
+  
+  
+  return(df)
+}
+
+
+# Older version of predict(...) to be removed 
+old_predict <- function(object, df = NULL, ...) {
```

Comment:
go ahead and just remove this now, I'd like to see how things look without the old functionality.